imgee/tasks.py

```diff
@@ -1,35 +1,83 @@
+import re
 import redis
-from imgee import app
+
+
+class InvalidRedisQueryException(Exception):
+    pass
 
 
 class TaskRegistry(object):
-    def __init__(self, name='default', connection=None):
-        self.connection = redis.from_url(connection) if connection else None
+    def __init__(self, name='default', connection=None, app=None):
+        if app:
+            self.init_app(name, connection)
+
+    def init_app(self, name='default', connection=None):
```

Comment:
`init_app` should not take any parameters apart from `app`. The other parameters go directly to `__init__`.